### DISCUSSION TITLE
Handling of modified event for selected items

### DISCUSSION THREAD

**Discussion by VikasAgarwal1984** [fabricjs/fabric.js#8325] [General]
Demo:
https://stackblitz.com/edit/angular-ivy-8mpwd1?file=src%2Fapp%2Fapp.component.ts

With:
http://fabricjs.com/v5-breaking-changes
IEvent.target property is removed on selection:created event for canvas object. 

Objective:
Handle modified event for the selected items to record position change.
As part of Demo sample above, Modified event is already defined for the individual Groups.

But user selecting bunch of Objects (groups), create a container group G1*, moving group G1* of selected items does not fire Modified event of individual Objects when Drag stops.
To handle this situation, target.on("modified", (e: IEvent) => {....}) handles the group modified event, where target is iEvent.target on selection:created event.

As per suggestion on http://fabricjs.com/v5-breaking-changes:
What to do
In the same events you can look at selected and deselected properties that contains an array of events. The first item of those array is what was referenced on those properties.

It is not clear how handle this situation.
I could possible find 2 ways:
1) 
const target = this.canvas.getActiveObject();

2) 
const target = ev.selected[0].group;

to define modified event on the group object.

Please provide suggestion on what is the right approach for the use case.

**Comment by ShaMan123** [comment]
This is a limitation that I have solved in v6 #7670 and specifically #7882

You need to access the active selection after it is created since currently canvas doesn't hold a constant ref to it.
Then append event listeners to and do the magic.
Pay attention to the fact that objects in a group are transformed relative to the group's plane. In v5 it is a nightmare trying to get something to work.
I suggest you look at the util `sendObjectToPlane`

**Reply by VikasAgarwal1984** [reply]
Thanks. I recently started using fabricjs.
My next question would have been on [7882](https://github.com/fabricjs/fabric.js/pull/7882)

**Reply by ShaMan123** [reply]
There is the v6-alpha branch that you can play with but master has diverged from it and I will drop some things from there.

**Comment by ShaMan123** [comment]
BTW I see you are an angular dev
I am looking for someone to polish an angular demo for fabric #8172

**Reply by VikasAgarwal1984** [reply]
i can plan to see this weekend i.e. 8th Oct. If this is okay.

**Reply by ShaMan123** [reply]
Great

**Reply by VikasAgarwal1984** [reply]
What are you exactly looking, seems templates/kitchensink is based on angular 1.6.9?
Should we have a demo for angular 14?

**Reply by ShaMan123** [reply]
Yes
I would like to refactor the demo to use the latest version

**Reply by ShaMan123** [reply]
And drop parcel

**Reply by ShaMan123** [reply]
It can be refactored to use a different framework
I don't mind
